core_test.go

```diff
@@ -23,6 +23,7 @@ func TestNew(t *testing.T) {
 	)
 
 	if c, err = testInit(); err != nil {
+		testTeardown(c)
```

Comment:
should probably use `t.AddHook(testTeardown)` inside of `testInit`, rather than having to remember that you need to call `testTeardown` everywhere.